nodes/fabric.js

```diff
@@ -121,32 +147,166 @@ module.exports = function (RED) {
+        }
+        if (!isNaN(endBlock)) {
+            options.endBlock = endBlock;
+            //https://jira.hyperledger.org/browse/FABN-1207
+            //Required due to bug
+            options.disconnect = false;
+        }
+        let event = null;
+        let eventList = [];
+        timeout = timeout === 'true' ? true : false;
+        if (timeout) {
+            var eventTimeout = setTimeout(() => {
+                if (event) {
+                    eventHub.unregisterChaincodeEvent(event);
+                    node.log('Unregistered chaincode event');
+                    msg.payload = eventList;
+                    node.send(msg);
+                }
+                node.log('Timed out for chaincode event (Expected)');
+            }, 2000);
+        }
+
+        node.log('Event listener options: ' + JSON.stringify(options));
+        event = eventHub.registerChaincodeEvent(chaincodeName, msg.payload.eventName, (event, blockNumber, txid, status) => {
+            var eventPayload = {
+                payload: event.payload.toString('utf8'),
+                blockNumber: blockNumber,
+                txid: txid,
+                status: status
+            };
+            // refresh timeout because we want ALL the events in the block interval
+            // not only the ones in the time interval
+            if (timeout) {
+                eventList.push(eventPayload);
+                eventTimeout.refresh();
+            } else {
+                msg.payload = eventPayload;
+                node.send(msg);
+            }
+            node.status({});
+        }, (error) => {
+            console.log(error);
+            node.error(error, msg);
+        }, options);
+        eventHub.connect(true);
+        node.log('Registered event listener');
+    }
+
+
+    /**
+     *
+     * @param {object} channel the channel object
+     * @param {string} blockNumber the number of the block to get
+     * @returns {PromiseLike<Contract | never>} promise
+     */
+    async function queryBlock(channel, blockNumber) {
+        return await channel.queryBlock(blockNumber);
+    }
+
+    /**
+     *
+     * @param {object} channel the channel object
+     * @param {string} transactionId the identifier of the transaction to get
+     * @returns {PromiseLike<Contract | never>} promise
+     */
+    async function queryTransaction(channel, transactionId) {
+        return await channel.queryTransaction(transactionId);
+    }
+
+    /**
+     *
+     * @param {string} identityName identityName
+     * @param {string} channelName channel
+     * @param {string} orgName name of the organization
+     * @param {string} peerName name of the peer
+     * @param {object} connection the connection config
+     * @param {string} walletLocation the wallet location (the files, not the folder)
+     * @returns {PromiseLike<Contract | never>} promise
+     */
+    async function connectToPeer(identityName, channelName,
```

Comment:
you don't need this function. You can get the channel from the network. You don't need the peer as you don't need to create a new event hub you can just get them like the subscribeToEvents function does, and you already have the identity from the wallet

Reply:
Correct. I'll remove it and use yours.